packages/client/src/components/SafeContacts.js

```diff
@@ -1,43 +1,203 @@
-import React from "react";
-import { Person } from "./Svg";
-import { useClipboard } from "../hooks";
+import React, { useState } from "react";
+import { isEmpty } from "lodash";
+import InputAddress from "./InputAddress";
+import InputText from "./InputText";
+import { Copy, Plus } from "./Svg";
+import { useClipboard, useContacts } from "../hooks";
+import { useModalContext } from "contexts";
 
-function SafeContacts({ safeOwners = [] }) {
+function EditContactModal({ contacts, contact, onConfirm, confirmText, closeModal }) {
+  const [currentAddr, setCurrentAddr] = useState(contact.address);
+  const [currentName, setCurrentName] = useState(contact.name);
+  const [addressValid, setAddressValid] = useState(Boolean(contact.address));
+  const [addressNew, setAddressNew] = useState(true);
+  const onAddressChange = ({ value, isValid }) => {
+    setCurrentAddr(value);
+    setAddressValid(isValid);
+    setAddressNew(isEmpty(contacts) || contacts.every(c => c.address !== value));
+  };
+  const onNameChange = ({ value }) => setCurrentName(value);
+  const canConfirm = addressNew && addressValid;
+  const updateBtnClasses = ["button", "flex-1", "p-4", canConfirm ? "is-link" : ""];
+
+  return (
+    <div className="py-5 has-text-black">
+      <div className="column is-flex is-flex-direction-column is-full px-5 py-0">
+        <div>
+          <p className="has-text-grey">Address<span className="has-text-red">*</span></p>
+          <InputAddress 
+            value={currentAddr} 
+            isValid={addressValid} 
+            onChange={onAddressChange} />
```

Comment:
seems it will always allow address change? For "Edit Contact", Address field need to be read-only, only Name can be updated. https://www.figma.com/file/sKtcz6ytBmNNDNW7XU8ea5/Vessel%3A-User-Experience?node-id=330%3A188312

Reply:
yeah seems ok to allow user to edit Address too, as long as it's a valid address. @almendili what do you think?

Reply:
confirmed with Jose that we should allow Edit Address too. 👍 @almendili, could you help update figma design to reflect the same? Thanks!